lib/seadsa/AllocWrapInfo.cc

```diff
@@ -154,30 +172,32 @@ bool AllocWrapInfo::findWrappers(Module &M, std::set<std::string> &fn_names) {
 /**
    Returns true if src flows into dst
  */
-bool AllocWrapInfo::flowsFrom(Value *dst, Value *src) {
+bool AllocWrapInfo::flowsFrom(Value *dst, Value *src, Pass *P) const {
   if (dst == src)
     return true;
 
   else if (ReturnInst *Ret = dyn_cast<ReturnInst>(dst)) {
-    return flowsFrom(Ret->getReturnValue(), src);
+    return flowsFrom(Ret->getReturnValue(), src, P);
   }
 
   else if (PHINode *PN = dyn_cast<PHINode>(dst)) {
     Function *F = PN->getParent()->getParent();
-    LoopInfo &LI = getAnalysis<LoopInfoWrapperPass>(*F).getLoopInfo();
+    // FIXME: we should use here getAnalysisIfAvailable. However, this
+    // method does not take a function as parameter.
+    LoopInfo &LI = P->getAnalysis<LoopInfoWrapperPass>(*F).getLoopInfo();
```

Comment:
`P->getAnalysisIfAvailable...`

Reply:
still unresolved

Reply:
My solution is that all clients of `AllocWrapInfo` calls `getAnalysis<LoopInfoWrapperPass>` before calling any of its methods. I don't know a better way

Reply:
I now see the comment in code that is a response for my comment. 
We should pass a pointer to LoopWrapInfoPass instead of a pointer to Pass